### DISCUSSION TITLE
Applying a static filter to the query with `with([....])`

### DISCUSSION THREAD

**Discussion by TheFrankman** [spatie/laravel-query-builder#499] [Q&A]
Hello there,

I'm not sure if I am missing something obvious here, but i want to utilise the power for of the query builder but in one instance apply a static filter to the query.

The `PropertyEnquiry` collection has a relation to `Property` class (belongsTo) the property has a `user_id` I only want to return enquiries for properties that belong to the authenticated user : 

```
$enquiries = QueryBuilder::for(PropertyEnquiry::class)
            ->with(["property" => function ($q) {
                $q->where('property.user_id', auth()->user()->id);
            }])
            ->allowedFilters('property_id')
            ->allowedSorts('property_id', 'start_date', 'end_date')
            ->allowedIncludes('property', 'property.images')
            ->get();
```

It would appear that the `with` gets ignored all together from the query builder and i get all enquires regardless of property owner.

Any assistance would be appreciated.

**Comment by dominikb** [comment]
I've done some digging into the `QueryBuilder`.

When you specify your own `->with('property',...)` clause, it gets added to the underlying `QueryBuilder` instance variable `$eagerLoads`. The name of the relation is the key and the value is the constraint closure.

Specifying an `allowedInclude('property', ...)` also adds, as a final step, the relation to the same `$eagerLoads` array, but does not respect an existing entry.
https://github.com/spatie/laravel-query-builder/blob/810be496f7ab3b70e391a049f76b9900321e9be2/src/Includes/IncludedRelationship.php#L10-L31

This could be fixed by checking for existing eager-load-entries and chaining the constraint closures within the `IncludeRelationship::class`.

For the time being you could apply this logic in your code like so:

```php
$query = QueryBuilder::for(PropertyEnquiry::class)
            ->allowedFilters('property_id')
            ->allowedSorts('property_id', 'start_date', 'end_date')
            ->allowedIncludes('property', 'property.images');

        if (array_key_exists('property', $query->getEagerLoads())) {
            $query->with(["property" => function ($q) {
                $q->where('property.user_id', auth()->user()->id);
            }]);
        }

$enquiries = $query->get();
```

**Reply by TheFrankman** [reply]
Thanks for the tip

I also realised i was being a bit of a dummy as I found an alternative solution that achieves the same goal, however, when i tried to find this issue, it wasn't appearing in the search (or when i filtered by me) so couldn't update. I simply did the same query from within a `whereHas`

: 


```
    $enquiries = QueryBuilder::for(PropertyEnquiry::class)
            ->whereHas('property', function (Builder $query) {
                $query->where('user_id', auth()->user()->id);
            })
            ->allowedFilters('property_id')
            ->allowedSorts('property_id', 'start_date', 'end_date')
            ->allowedIncludes('property', 'property.images')
            ->get();

        return response()->json($enquiries);
```
